### ISSUE TITLE
[2014 ICLR] Network In Network

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#10]
기존의 conventional convolution neural network는 linear convolution filter - nonlinear activation function 구조를 stack하여 feature map을 뽑은 후, fully connected layer를 통해 classification을 함

이는 2가지 문제점을 가짐

1. Assumption: latent concepts are linearly separable or lie within convex sets
    - Conventional CNN의 convolution filter는 generalized linear model (GLM)로 해석할 수 있음
    - GLM은 latent concepts가 linearly separable할 경우 좋은 성능을 낼 수 있음
    - Maxout network가 linear function의 maximization을 통해 piecewise linear approximator로 해결하고자 했으나 이 또한 convex sets에 있다는 가정을 함
    - 몇몇 실제 데이터는 nonlinear manifold에 위치하기에 해당 assumption에 어긋남
    - 따라서 nonlinear function이 필요함

2. Fully connected layers
    - Fully connected layers는 overfitting하기 쉬워 model의 generalization ability를 저하시킴 (dropout regularization로 해결가능)
    - Fully connected layers에서 많은 parameter가 사용되어 학습하는데 시간이 오래걸림
    - 여러 fully connected layers로 인해 feature map이 변형되어 classification을 하기에, category와 feature map간의 correspondence가 떨어짐

저자들은 이에 대하여 다음과 같은 방법을 통해 문제를 해결하고자 했음
![image](https://user-images.githubusercontent.com/49643709/113291065-6c9d6d80-932d-11eb-85e7-472cd94b8605.png)

1. MLP convolution layers
    - Latent concepts에 prior distribution을 주지 않고 사용할 수 있는 universal function approximator
    - Radial basis network과 다르게 multilayer perceptron은 확장성, back-propagation이 가능하기에 사용
    - Convolution layer 뒤에 mlp layers를 stack하는 'micro network structure' (nonlinear activation function이 뒤에 붙음)
    - 연결 방식은 1*1 커널의 convolution layer라고 생각해도 됨 (cross channel parametric pooling)
    - 해당 structure이 general nonlinear function approximator이기에, abstraction ability 향상

![image](https://user-images.githubusercontent.com/49643709/113291227-b1c19f80-932d-11eb-99f3-2026d2d04bed.png)

2. Global average pooling
    - Feature map의 spatial average를 계산
    - 학습을 안하기에 overfitting issue가 없음
    - Spatial translation에 robust함
    - Category와 feature map 사이의 correspondence를 강제하기에 meaningful, interpretable함

**Comment by Jasonlee1995** [comment]
데이터셋 별 실험 결과는 다음과 같음

학습은 maxout network와 유사하다고 함
- Batch size : 128
- SGD with momentum 0.9, wegith decay 0.00001 (learning rate unknown)
- Learning rate scheduler : lower by a scale of 10 when training set accuracy stops improving

### CIFAR 10/100
![image](https://user-images.githubusercontent.com/49643709/113291269-bede8e80-932d-11eb-8267-4799b26c0203.png)
![image](https://user-images.githubusercontent.com/49643709/113291281-c30aac00-932d-11eb-9732-30e5e8676a6c.png)
- Augmentation : global contrast normalization, ZCA whitening, translation, horizontal flipping

### SVHN
![image](https://user-images.githubusercontent.com/49643709/113291289-c6059c80-932d-11eb-947f-d2683bb98b47.png)
- Augmentation : local contrast normalization

### MNIST
![image](https://user-images.githubusercontent.com/49643709/113291300-c9008d00-932d-11eb-9f62-5492337c969a.png)
- No data augmentation
